File: Scripts/micro_dao/bairro.py
from micro_dao.objeto_modelo import ObjetoModelo


class Bairro(ObjetoModelo):
    _id: int
    _nome: str
    _fk_cidade: int

    def __init__(self, nome_bairro: str, fk_cidade: int):
        self.set_nome(nome_bairro)
        self.set_fk_cidade(fk_cidade)

    def get_id(self) -> int:
        return self._id

    def set_id(self, id: int):
        self._id = id
        return self

    def get_nome(self) -> str:
        return self._nome

    def set_nome(self, nome_cidade: str):
        self._nome = nome_cidade
        return self

    def get_fk_cidade(self) -> int:
        return self._fk_cidade

    def set_fk_cidade(self, fk_cidade: int):
        self._fk_cidade = fk_cidade
        return self

File: Projeto_Java/src/Dominio/Entidades/Bairro.java
package Dominio.Entidades;

import javax.persistence.*;

@Entity
@Table(name = "bairro")
public class Bairro {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "nome", length = 60)
    private String nome;

    @Transient
    private int fkCidade;

    @OneToOne(fetch = FetchType.LAZY, targetEntity = Cidade.class)
    @JoinColumn(name = "fk_cidade", referencedColumnName = "id")
    private Cidade cidade;


    public Bairro() {}

    public Bairro(String nome, int fkCidade){
        setNome(nome);
        setFkCidade(fkCidade);
    }

    public Bairro(int id, String nome, int fkCidade){
        setId(id);
        setNome(nome);
        setFkCidade(fkCidade);
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getFkCidade() {
        return fkCidade;
    }

    public void setFkCidade(int fkCidade) {
        this.fkCidade = fkCidade;
    }

    public Cidade getCidade() {
        return cidade;
    }

    public void setCidade(Cidade cidade) {
        this.cidade = cidade;
    }

    //Métodos para uso do Hibernate
    @PostLoad
    private void setFkCidade() {
        this.setFkCidade(getCidade().getId());
    }
}
